core/types/rollup_cost.go

```diff
@@ -259,11 +295,26 @@ func intToScaledFloat(scalar *big.Int) *big.Float {
 
 // extractL1GasParams extracts the gas parameters necessary to compute gas costs from L1 block info
 func extractL1GasParams(config *params.ChainConfig, time uint64, data []byte) (gasParams, error) {
-	// edge case: for the very first Ecotone block we still need to use the Bedrock
-	// function. We detect this edge case by seeing if the function selector is the old one
-	// If so, fall through to the pre-ecotone format
-	// Both Ecotone and Fjord use the same function selector
-	if config.IsEcotone(time) && len(data) >= 4 && !bytes.Equal(data[0:4], BedrockL1AttributesSelector) {
+	if config.IsIsthmus(time) && len(data) >= 4 && !bytes.Equal(data[0:4], EcotoneL1AttributesSelector) {
+		// edge case: for the very first Isthmus block we still need to use the Ecotone
```

Comment:
nit
```suggestion
		// edge case: for the very first Isthmus block we still need to use the Fjord
```

Reply:
Ecotone is correct there, we go from `extractL1GasParamsPostEcotone` to `extractL1GasParamsPostIsthmus`